### PR TITLE
[TEST DNM] memory slots: enable dynamic slots allocation

### PR DESCRIPTION
**Pull request by piotrkmx**
Enables dynamic slots allocation.

Signed-off-by: Piotr Kmiecik <piotrx.kmiecik@intel.com>

### CODE REVIEW

**Comment by marc-hb** [comment]
These sparse warnings found on both TGL and MTL are new:

https://github.com/thesofproject/sof/actions/runs/3891972074/jobs/6642841867

```
/zep_workspace/zephyr/drivers/memory_window/dynamic_slots.c:23:31: warning: incorrect type in initializer (different address spaces)
/zep_workspace/zephyr/drivers/memory_window/dynamic_slots.c:37:26: warning: incorrect type in initializer (different address spaces)
/zep_workspace/zephyr/drivers/memory_window/dynamic_slots.c:45:57: warning: incorrect type in argument 1 (different address spaces)
/zep_workspace/zephyr/drivers/memory_window/dynamic_slots.c:108:26: warning: incorrect type in initializer (different address spaces)
/zep_workspace/zephyr/drivers/memory_window/dynamic_slots.c:112:57: warning: incorrect type in argument 1 (different address spaces)
/zep_workspace/zephyr/drivers/memory_window/dynamic_slots.c:154:31: warning: incorrect type in initializer (different address spaces)
/zep_workspace/zephyr/drivers/memory_window/dynamic_slots.c:158:57: warning: incorrect type in argument 1 (different address spaces)
/zep_workspace/zephyr/drivers/telemetry/telemetry_intel.c:23:38: warning: incorrect type in assignment (different address spaces)
```

MTL has additional warnings which have always been there (very confusing sorry) but TGL has been sparse-clean for a long time.

File: west.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ manifest:
 
     - name: zephyr
       repo-path: zephyr
-      revision: e4fcb32451c587a3e4ba7f8bf3fc602b16f9652b
+      revision: cb86242cd1e61c32daee4bc614b9ee87d0a07b89
```
Comment (marc-hb) [inline west.yml:46]: For the record this cb86242cd1e
commit is testing telemetry zephyr PR https://github.com/zephyrproject-rtos/zephyr/pull/53057

Comment (lgirdwood) [reply west.yml:46]: @marc-hb @piotrkmx are we blocking on above Zephyr PR to merge ?

**Comment by marc-hb** [comment]
> @marc-hb @piotrkmx are we blocking on above Zephyr PR to merge ?

I believe this is indeed only a **test PR**, testing a Zephyr PR in SOF CI. So it's not meant for review. The title, description and comments should have made that more obvious.

To be fair it is a Github draft, meaning it's _not_ supposed to be reviewed unless explicitly requested to do so :-) 

To exclude drafts: https://github.com/thesofproject/sof/pulls?q=is%3Apr+is%3Aopen+-is%3Adraft

About half of the SOF PRs currently open are drafts

**Comment by lyakh** [comment]
> These sparse warnings found on both TGL and MTL are new:

@piotrkmx yeah, some clean up is needed. `__sparse_cache` indicates to sparse that the address belongs to a "special" address space - a `__cache` address space, and is thus incompatible with the default address space. So, all assignments must happen between the same AS pointers, also function parameters must match. Your PR mixes them up a bit. I'll comment in the PR.

**Pull request closed without merging**